util/util.cpp

```diff
@@ -106,41 +518,16 @@ double getSensorDbusTemp(std::string sensorDbusPath, bool unitMilli)
 
     if (spofEnabled)
     {
-        // If sensor functional property is false, set NaN to dbus and return NaN.
-        if (!dbus::SDBusPlus::checkFunctionalProperty(bus,
-                                                      service,
-                                                      sensorDbusPath))
-        {
-            return value;
-        }
+        std::string alarmText = checkSensorAlarmProperties(bus, service, sensorDbusPath);
```

Comment:
Hi Krellan, sorry for the late review. Thanks to improve this daemon.
After testing, the behavior is correct in normal situation, but seems there is some problem when sensor fail.
When I trigger sensor warning/critical high, the value will set as NaN and fans will into fanfail duty in the beginning, but after few seconds, it will back to normal duty cycle.
If I understand correctly, does there need "setValueCache" when detecting alarmText not empty?

Reply:
For the concern over warning/critical thresholds getting triggered, perhaps catching "CriticalHighAlarmAsserted" and "WarningHighAlarmAsserted" signals would help?

Reply:
I see, I think the problem is in my setup, the phosphor-hwmon sensors are not configured with thresholds yet.

Do you see DBus signals that look like the following emitted from a phosphor-hwmon sensor, when a threshold is exceeded? (These can be seen with `dbus-monitor --system`)
```
signal time=1636003048.639621 sender=:1.47 -> destination=(null destination) serial=33
path=/xyz/openbmc_project/sensors/utilization/CPU;
interface=xyz.openbmc_project.Sensor.Threshold.Critical;
member=CriticalHighAlarmAsserted
   double 22.1945
```

It looks like dbus-sensors does not make use of those signals yet (because methods like `sdbusplus::xyz::openbmc_project::Sensor::Threshold::server::Warning::sdbusplus::xyz::openbmc_project::Sensor::Threshold::server::Warning` do not appear in dbus-sensors). It does appear in phosphor-hwmon though.

So I guess for this optimization to work with dbus-sensors correctly, dbus-sensors needs to be updated to emit those signals when thresholds are crossed.

Reply:
Update: it seems we don't need the `Asserted` signals because the same assertion information can be checked in the `PropertyChanged` signal handlers. We also checked a few dbus dumps with and without the patch applied and we can see the program functioning as intended and the number of GetAll method calls greatly reduced with the improvement.